SimG4Core/CustomPhysics/src/G4SQInelasticProcess.cc

```diff
@@ -0,0 +1,241 @@
+
+#include "SimG4Core/CustomPhysics/interface/G4SQInelasticProcess.h"
+#include "SimG4Core/CustomPhysics/interface/G4SQInelasticCrossSection.h"
+#include "SimG4Core/CustomPhysics/interface/G4SQ.h"
+
+#include "G4Types.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4HadProjectile.hh"
+#include "G4ElementVector.hh"
+#include "G4Track.hh"
+#include "G4Step.hh"
+#include "G4Element.hh"
+#include "G4ParticleChange.hh"
+#include "G4NucleiProperties.hh"
+#include "G4Nucleus.hh"
+
+#include "G4HadronicException.hh"
+#include "G4HadronicProcessStore.hh"
+#include "G4HadronicInteraction.hh"
+
+#include "G4ParticleDefinition.hh"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+G4SQInelasticProcess::G4SQInelasticProcess(double mass, const G4String& processName)
+    : G4HadronicProcess(processName, fHadronic) {
+  AddDataSet(new G4SQInelasticCrossSection(mass));
+  theParticle = G4SQ::SQ(mass);
+}
+
+G4SQInelasticProcess::~G4SQInelasticProcess() {}
+
+G4bool G4SQInelasticProcess::IsApplicable(const G4ParticleDefinition& aP) {
+  return theParticle->GetParticleType() == aP.GetParticleType();
+}
+
+G4VParticleChange* G4SQInelasticProcess::PostStepDoIt(const G4Track& aTrack, const G4Step& aStep) {
+  edm::LogInfo("G4SQInelasticProcess::PostStepDoIt")
+      << "Particle is going to interact at position" << aTrack.GetPosition() / cm
+      << " momentumdirection eta: " << aTrack.GetMomentumDirection().eta()
+      << " interacting in material : " << aTrack.GetMaterial() << std::endl;
+
+  // if primary is not Alive then do nothing
+  theTotalResult->Clear();
+  theTotalResult->Initialize(aTrack);
+  theTotalResult->ProposeWeight(aTrack.GetWeight());
+  if (aTrack.GetTrackStatus() != fAlive) {
```

Comment:
This cannot happens

Reply:
Ok, anyway whole method removed as per your later comment.